### PR TITLE
feature(menus): elgg_register_title_button() can now check entity typ…

### PR DESCRIPTION
**Pull request by hypeJunction**
…e and subtype

This adds more granularity to elgg_register_title_button() allowing to check container
permissions for a specific entity type and subtype before registering a title menu item

### CODE REVIEW

**Comment by jdalsem** [comment]
LGTM

File: engine/lib/navigation.php [hunk 1]
```diff
@@ -204,33 +204,34 @@ function elgg_get_menu_item($menu_name, $item_name) {
...
-				'text' => elgg_echo("$handler:$name"),
-				'link_class' => 'elgg-button elgg-button-action',
-			));
-		}
+	$owner = elgg_get_page_owner_entity();
+	if (!$owner) {
+		// noone owns the page so this is probably an all site list page
+		$owner = elgg_get_logged_in_user_entity();
+	}
+	if (!$owner || !$owner->canWriteToContainer(0, $entity_type, $entity_subtype)) {
```
Comment (mrclay) [inline engine/lib/navigation.php]: This assumes the button is for writing to the container, which may not be true.

Comment (mrclay) [inline engine/lib/navigation.php]: Oops, it already does this.

**Comment by mrclay** [comment]
LGTM

**Pull request merged**